contracts/access/TimelockController.sol

```diff
@@ -0,0 +1,294 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.6.0;
+pragma experimental ABIEncoderV2;
+
+import "./../math/SafeMath.sol";
+import "./AccessControl.sol";
+
+/**
+ * @dev Contract module which acts as a timelocked controller. When set as the
+ * owner of an `Ownable` smart contract, it enforces a timelock on all
+ * `onlyOwner` maintenance operations. This gives time for users of the
+ * controlled contract to exit before a potentially dangerous maintenance
+ * operation is applied.
+ *
+ *
+ * This contract is designed to be self administered, meaning it should be its
+ * own administrator. The proposer (resp executor) role is in charge of
+ * proposing (resp executing) operation. A common use case is to position this
+ * {TimelockController} as the owner of a smart contract, with a multisig or a
+ * DAO as the sole proposer. Once at least one proposer and one executor have
+ * been appointed, self-administration can be enable using the `makeLive`
+ * function.
+ */
+contract TimelockController is AccessControl {
+
+    bytes32 public constant ADMIN_ROLE = keccak256("ADMIN_ROLE");
+    bytes32 public constant PROPOSER_ROLE = keccak256("PROPOSER_ROLE");
+    bytes32 public constant EXECUTOR_ROLE = keccak256("EXECUTOR_ROLE");
+    uint256 internal constant _DONE_TIMESTAMP = uint256(1);
+
+    mapping(bytes32 => uint256) private _timestamps;
+    uint256 private _minDelay;
+
+    /**
+     * @dev Emitted when a call is scheduled as part of operation `id`.
+     */
+    event CallScheduled(bytes32 indexed id, uint256 indexed index, address target, uint256 value, bytes data, bytes32 predecessor);
```

Comment:
I think this event should also contain the `delay` number.

Reply:
delay or timestamp ?

Reply:
Maybe timestamp is the more useful number.

Reply:
The event is in a transaction. The transaction has a block number. The block contains a timestamp.

In TheGraph it is as easy as doing `event.block.timestamp`

Reply:
That is for the current timestamp. If you are asking is the operation timestamp, this can either be recomputed (block timestamp + delay) or by querying the Timelock `getTimestamp` function with the id present in the event:

https://github.com/Amxx/openzeppelin-contracts-subgraphs/blob/master/templates/TimelockController/src/TimelockController.ts#L40

Reply:
Looking at the JSON RPC interface: if you want to get the timestamp based on the delay contained in an event, you need an extra RPC call to get the timestamp for the event's blockHash. But conversely, if the event contains the timestamp, the same extra RPC call is required to compute the delay.

I think it's reasonable that interfaces will want to display both values, and the event should only contain one of them. Let's keep it as is and emit only the delay.